### PR TITLE
Fix status-reconciler image name

### PR DESCRIPTION
**Pull request by alvaroaleman**
Also bumped the tag as the old one doesn't seem to exist anymore (taken from prow/cluster/deck_deployment.yaml)

/assign @stevekuznetsov

### CODE REVIEW

File: prow/cluster/statusreconciler_deployment.yaml [hunk 1]
```diff
@@ -30,7 +30,7 @@ spec:
       terminationGracePeriodSeconds: 180
       containers:
       - name: statusreconciler
-        image: gcr.io/k8s-prow/statusreconciler:v20181129-31c544b
+        image: gcr.io/k8s-prow/status-reconciler:v20181220-899fabe
```
Comment (fejta) [inline prow/cluster/statusreconciler_deployment.yaml]: Please use the same tag as the rest of prow: https://github.com/kubernetes/test-infra/pull/10673

**Comment by fejta** [comment]
/hold

Can you also add it to https://github.com/kubernetes/test-infra/blob/85c2788d9ddac6fed105a6f60a85ae21a8e1e6d6/prow/cluster/BUILD.bazel#L8-L38 so we actually deploy it (after which we'll get errors I think if the image is bad)

**Comment by alvaroaleman** [comment]
@fejta  regarding deploying the status_reconciler there was a discussion on Slack about that: 
https://kubernetes.slack.com/archives/CDECRSC5U/p1546463409045400?thread_ts=1546463228.045100&cid=CDECRSC5U

Are you sure this is wanted? Maybe @cjwagner can comment on that as well?

**Comment by cjwagner** [comment]
Yeah, I don't think that the status-reconciler can handle the scale of the kubernetes org in its current state.

**Comment by fejta** [comment]
Can we run it once an hour/day or something instead?

**Comment by stevekuznetsov** [comment]
The issue is that if you add a new presubmit to `kubernetes/kubernetes`, it will trigger for all open PRs. But I guess I don't really know what you guys do in that case  today anyway, just end up in a broken state?

**Comment by stevekuznetsov** [comment]
I will say ~25% of your open PRs need a rebase, so the jobs would insta-fail there anyway and not cost too many resources.

**Comment by stevekuznetsov** [comment]
Depending on how often you guys add brand-new blocking presubmits to `kubernetes/kubernetes`, you should be totally fine to add this and make it useful for all other repos that you manage.

For all jobs except those touching external state (AWS/EC2/EKS) you should be fine to have limits/requests on your Pods and trust the k8s scheduler to do it's thing?

**Comment by BenTheElder** [comment]
> The issue is that if you add a new presubmit to kubernetes/kubernetes, it will trigger for all open PRs. But I guess I don't really know what you guys do in that case today anyway, just end up in a broken state?

yeah, people have to /test ... it's not great, but not quite broken. a push will also trigger it.

for removing, we run the manual migrator

**Comment by BenTheElder** [comment]
> For all jobs except those touching external state (AWS/EC2/EKS) you should be fine to have limits/requests on your Pods and trust the k8s scheduler to do it's thing?

only if _all_ of the pods do this, which is unlikely because it's a pain for job authors to figure this out currently.

**Comment by alvaroaleman** [comment]
@fejta @cjwagner Are you fine with me removing the second commit from this PR that activates the `status-reconciler` as there doesn't seem consensus around this?

I'd really like to get the first commit merged because for someone trying to deploy the `status-reconciler`, that broken image name will waste quite some time

**Comment by fejta** [comment]
Yes let's punt enabling it to another commit, but we should definitely build it.

**Comment by alvaroaleman** [comment]
@fejta Not sure I understand your comment, isn't it already getting built? Would this PR be good to go after I remove the changes to `prow/cluster/BUILD.bazel`  from it?

**Comment by fejta** [comment]
yes that's what I meant

**Comment by alvaroaleman** [comment]
Removed it, PTAL

File: prow/cluster/statusreconciler_deployment.yaml [hunk 2]
```diff
@@ -30,7 +30,7 @@ spec:
       terminationGracePeriodSeconds: 180
       containers:
       - name: statusreconciler
-        image: gcr.io/k8s-prow/statusreconciler:v20181129-31c544b
+        image: gcr.io/k8s-prow/status-reconciler:v20190109-6897efb
```
Comment (cjwagner) [inline prow/cluster/statusreconciler_deployment.yaml]: The mixing of `statusreconciler` and `status-reconciler` throughout filenames, config files, and BUILD files is :-1: 
Lets not hold up this fix though.
/lgtm

**Pull request merged**